File: PhpWebApi/app/Models/UserExtend.php
<?php

namespace App\Models;

/**
 * 用户扩展表
 */
class UserExtend extends BaseModel
{
    // 与模型关联的数据表名
    protected $table = 'user_extend';
    // 与数据表关联的主键
    protected $primaryKey = 'id';
    // 指明模型的ID是否自动递增
    public $incrementing = true;
    // 自动递增ID的数据类型
    protected $keyType = 'integer';
    // 指示模型是否主动维护时间戳
    public $timestamps = false;

    // 可修改的表字段
    protected $fillable = [
        'uid',
        'birthday',
        'height',
        'weight',
    ];
}

File: JavaWebApi/src/main/java/com/laixhe/webapi/entity/UserExtend.java
package com.laixhe.webapi.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.persistence.UniqueConstraint;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * 用户扩展 (对应 Go models/user_extend.go)
 */
@Entity
@Table(name = "user_extend", uniqueConstraints = @UniqueConstraint(name = "user_extend_uid_idx", columnNames = "uid"))
@Getter
@Setter
@NoArgsConstructor
public class UserExtend {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "uid", nullable = false)
    private Integer uid;

    /** 生日(年月日) */
    @Column(name = "birthday", nullable = false)
    private Integer birthday = 0;

    /** 身高(cm) */
    @Column(name = "height", nullable = false)
    private Integer height = 0;

    /** 体重(kg) */
    @Column(name = "weight", nullable = false)
    private Integer weight = 0;
}
